registry/api/errcode/doc.go

```diff
@@ -0,0 +1,64 @@
+// Package errcode defines an error management package.
+//
+// The `errcode` package allows for each error to be typed and include all
+// information necessary to have further processing done on them if necessary.
+//
+// Use of this package is defined by the following flow:
+// * Each error is registered with the errcode package via the `Register()`
+// function. The `Register()` function takes a `group` name and an
+// `ErrorDescriptor` structure. The `group` name allows for errors to be
+// associated with a particular component, or any other grouping mechanism
+// that may have meaning to the code registering the error.  The
+// `ErrorDescriptor` describes the error itself. See below for more
+// information. The `Register()` function will return an `ErrorCode`
+// that uniquely identifies the register error.
+//
+// * Once an error is registered, the return `ErrorCode` can be used just like
+// any other golang `error` type.
+//
+// * If a particular error needs to have additional information or processing
+// performed then the `WithArgs()` and `WithDetails()` functions are available.
+// The `WithArgs()` function allows for the code generating the error to
+// specify the substitution values of the `%s` variables in the error's
+// message. `WithDetails()` allows for the specification of any additional
+// information that may need to be provided to the end-user for this particular
+// error.  In both cases, the functions will return an `Error` resource
+// which extends the `ErrorCode` resource with the additonal information.
+//
+// The package consists of three main resource types:
+//
+// * ErrorCode
+// ErrorCode is a unique (numerical) identifier for a particular error
```

Comment:
It might help if we can decide on exactly how people are supposed to use this bit of information.  Initially, I thought this was a unique "ID" that people could then use to do some kind of look-up to find more info about the error. For example, in all of the IBM products we've been involved in, each error generated had a unique ID that end users could use to search the docs for more info. And this was true irrespective of which language the error was displaying. This meant that an english speaking person could still figure out what a chinese error message was all about.

However, I'm not sure that we're on the same page w.r.t. how this string is meant to be used.  What were your thoughts behind it?  If we don't have a clear understanding how of the end-user is meant to use this bit of info (yet) then I'm inclined to remove (or at least hide it) until we can justify it. Likewise, if its just for docker devs then we need to explain how they can use it too.